upup/pkg/fi/cloudup/defaults.go

```diff
@@ -227,3 +238,36 @@ func incrementIP(ip net.IP, cidr string) (string, error) {
 	}
 	return ip.String(), nil
 }
+
+// setupEctdVersion enables the supported Etcd version, if not set.
+func setupEctdVersion(cluster *kops.Cluster) error {
+	parsedVersion, err := util.ParseKubernetesVersion(cluster.Spec.KubernetesVersion)
+	if err != nil {
+		return fmt.Errorf("error parsing version: %v", cluster.Spec.KubernetesVersion)
+	}
+
+	version := ""
+	// as more versions are release we can add more if else statements here to set version.
+	if util.IsKubernetesGTE("1.9.0", *parsedVersion) {
+		// https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md#external-dependencies
+		// lists 3.1.10 as the supported version for 1.9.x
+		version = "3.1.10"
+	}
+
+	// TODO - do we want to do this with new 1.8 clusters?
+	// else if util.IsKubernetesGTE("1.8.0", *parsedVersion) {
+	// 	version = "3.0.17"
+	// }
+
+	if version == "" {
+		return nil
+	}
+
+	for _, etcd := range cluster.Spec.EtcdClusters {
+		if etcd.Version == "" {
+			etcd.Version = version
```

Comment:
If version is being updated as per previous comment, we would also need to set the image location:
```
etcd.Image = "gcr.io/etcd-development/etcd:v3.3.0"
```

Reply:
Also enable etcd TLS by default?